// @clds/typography/dist/esm/apps/typographyMixin.js
import { css } from 'styled-components';
import { appsTheme } from './theme';
export const getTypographyRuleMixins = (size, type) => {
    const sizeBranch = appsTheme[size];
    if (!size || !type) {
        throw new Error(`Apps typography: type ${type} and size ${size} is not valid configuration`);
    }
    if (!(type in sizeBranch)) {
        throw new Error(`Apps typography: ${type} variant is not supported for ${size} typography`);
    }
    const variantBranch = sizeBranch[type];
    return {
        fontSize: sizeBranch.size,
        lineHeight: sizeBranch.lineHeight,
        color: variantBranch.color,
        fontWeight: variantBranch.weight,
    };
};
const isProps = (arg1, arg2) => {
    return arg1 !== undefined && arg2 === undefined;
};
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function typographyMixin(arg1, arg2) {
    const size = isProps(arg1, arg2) ? arg1.size : arg1;
    const type = isProps(arg1, arg2) ? arg1.type : arg2;
    const mixins = getTypographyRuleMixins(size, type);
    return css `
    font-size: ${mixins.fontSize};
    line-height: ${mixins.lineHeight};
    color: ${mixins.color};
    font-weight: ${mixins.fontWeight};
  `;
}
